ocaml/xapi/vgpuops.ml

```diff
@@ -69,23 +70,31 @@ let allocate_vgpu_to_gpu ~__context vgpu available_pgpus =
   in
   choose_pgpu sorted_pgpus
 
-let create_passthrough_vgpu ~__context ~vm vgpu available_pgpus =
-  debug "Creating passthrough VGPUs";
-  match allocate_vgpu_to_gpu ~__context vgpu available_pgpus with
-  | None ->
+let fail_creation vm vgpu =
+  match vgpu.requires_passthrough with
+  | None | Some `VF ->
+    raise (Api_errors.Server_error (Api_errors.vm_requires_vgpu, [
+        Ref.string_of vm;
+        Ref.string_of vgpu.gpu_group_ref;
+        Ref.string_of vgpu.type_ref
+      ]))
+  | Some `PF ->
     raise (Api_errors.Server_error (Api_errors.vm_requires_gpu, [
         Ref.string_of vm;
         Ref.string_of vgpu.gpu_group_ref
       ]))
+
+let create_vgpu ~__context ~vm ~host vgpu =
```

Comment:
Having a `create_vgpu` function return `pgpu` looks weird on the face of it.
Right now I understand it and it makes sense, but for the sake of anyone trying to understand this code in future, let's have a comment saying what the function does and what it returns.